server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -80,9 +79,47 @@
      * @throws MessagingException when the message is not clonable
      */
     public static MailImpl duplicate(Mail mail) throws MessagingException {
-        return new MailImpl(mail, newName(mail));
+        return new MailImpl(mail, deriveNewName(mail.getName()));
     }
 
+    private static final java.util.Random random = new java.util.Random(); // Used
+    // to
+    // generate
+    // new
+    // mail
+    // names
+
+    /**
+     * Create a unique new primary key name for the given MailObject.
+     *
+     * @param currentName the mail to use as the basis for the new mail name
+     * @return a new name
+     */
+    @VisibleForTesting static String deriveNewName(String currentName) throws MessagingException {
+
+        // Checking if the original mail name is too long, perhaps because of a
+        // loop caused by a configuration error.
+        // it could cause a "null pointer exception" in AvalonMailRepository
+        // much
+        // harder to understand.
+        if (currentName.length() > 76) {
+            int count = 0;
+            int index = 0;
+            while ((index = currentName.indexOf('!', index + 1)) >= 0) {
+                count++;
+            }
+            // It looks like a configuration loop. It's better to stop.
+            if (count > 7) {
+                throw new MessagingException("Unable to create a new message name: too long." + " Possible loop in config.xml.");
```

Comment:
Is that really still necessary?

Reply:
Why would I changed that right now ? I mean, there's a piece of code with a behavior, I don't know how to test it and don't want to take the risk to break it. People wrote it for a good reason and it looks like it doesn't hurt. Let's way for a better understanding before changing that.